### PR TITLE
[query] topic api: support DELETE on /topic

### PR DESCRIPTION
**Pull request by schallert**
Users may need to delete m3msg topics.

**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
Added support for deleting m3msg topics in coordinator API
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
None
```

### CODE REVIEW

File: src/query/api/v1/handler/topic/get_test.go [hunk 1]
```diff
@@ -94,21 +87,3 @@ func TestTopicGetHandler(t *testing.T) {
 	resp = w.Result()
 	require.Equal(t, http.StatusNotFound, resp.StatusCode)
 }
-
-func validateEqualTopicProto(t *testing.T, this, other topicpb.Topic) {
```
Comment (benraskin92) [inline src/query/api/v1/handler/topic/get_test.go]: Meant to delete these tests?

Comment (schallert) [reply src/query/api/v1/handler/topic/get_test.go]: Yeah moved to common_test.go

Comment (benraskin92) [reply src/query/api/v1/handler/topic/get_test.go]: Ah sorry just saw

File: src/query/api/v1/handler/topic/common_test.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
```
Comment (benraskin92) [inline src/query/api/v1/handler/topic/common_test.go]: nit: `2019`

**Pull request merged**